tests/zero-downtime-scaling/results/1.1.0/1.1.0.md

```diff
@@ -0,0 +1,512 @@
+# Results
+
+## Versions
+
+Kubernetes:
+
+```text
+Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.5-gke.200",
+GitCommit:"f9aad8e51abb509136cb82b4a00cc3d77d3d70d9", GitTreeState:"clean", BuildDate:"2023-08-26T23:26:22Z",
+GoVersion:"go1.20.7 X:boringcrypto", Compiler:"gc", Platform:"linux/amd64"}
+```
+
+NGF:
+
+```text
+"version":"edge",
+"commit":"7a6e198c8b6e1e81058cd9ad0717cffa2cc77bb8",
+"date":"2023-12-07T22:27:22Z"
+```
+
+with NGINX:
+
+```text
+nginx version: nginx/1.25.3
+built by gcc 12.2.1 20220924 (Alpine 12.2.1_git20220924-r10)
+built with OpenSSL 3.1.3 19 Sep 2023 (running with OpenSSL 3.1.4 24 Oct 2023)
+```
+
+## Summary
+
+Results are similar with 1.0.0 test results. Did not encounter instances of
+NGF Pod readiness flagging under high load. Downtime and small errors/warnings still persisted.
+
+## 25 Node Cluster with anti affinity
+
+### Scale Up Gradually
+
+HTTP wrk output:
+
+```text
+Running 5m test @ http://cafe.example.com/coffee
+  2 threads and 2 connections
+  Thread Stats   Avg      Stdev     Max   +/- Stdev
+    Latency     3.31ms   42.03ms   1.04s    99.51%
+    Req/Sec     1.42k   120.96     1.70k    78.24%
+  Latency Distribution
+     50%  684.00us
+     75%  765.00us
+     90%    0.86ms
+     99%    1.54ms
+  725404 requests in 5.00m, 259.07MB read
+  Socket errors: connect 0, read 19, write 0, timeout 3
+Requests/sec:   2417.39
+Transfer/sec:      0.86MB
+```
+
+HTTP Graph:
+
+![gradual-up-http.png](25-node/gradual-up-http.png)
+
+HTTPS wrk output:
+
+```text
+Running 5m test @ https://cafe.example.com/tea
+  2 threads and 2 connections
+  Thread Stats   Avg      Stdev     Max   +/- Stdev
+    Latency     6.12ms   60.34ms   1.07s    99.04%
+    Req/Sec     1.25k   106.37     1.44k    84.47%
+  Latency Distribution
+     50%  775.00us
+     75%    0.87ms
+     90%    0.98ms
+     99%   39.75ms
+  561506 requests in 5.00m, 196.84MB read
+  Socket errors: connect 0, read 17, write 0, timeout 2
+Requests/sec:   1871.29
+Transfer/sec:    671.75KB
+```
+
+HTTPS graph:
+
+![gradual-up-https.png](25-node/gradual-up-https.png)
+
+Logs:
+
+- NGF error logs: none
+- NGINX error logs:
+
+  Handful of these errors and warnings:
+
+  ```text
+  2023/12/11 19:14:13 [warn] 77#77: *159408 upstream server temporarily disabled while connecting to upstream, client: 10.128.0.2, server: cafe.example.com, request: "GET /tea HTTP/1.1", upstream: "http://10.100.6.4:8080/tea", host: "cafe.example.com"
+  2023/12/11 19:14:13 [error] 77#77: *159408 upstream timed out (110: Operation timed out) while connecting to upstream, client: 10.128.0.2, server: cafe.example.com, request: "GET /tea HTTP/1.1", upstream: "http://10.100.6.4:8080/tea", host: "cafe.example.com"
+  ```
+
+- NGINX access logs:
+  - 1.3 million 200s
+
+### Scale Down Gradually
+
+HTTP wrk output:
```

Comment:
Should I mark that down anywhere?

Reply:
how about future improvements section? similar to https://github.com/nginxinc/nginx-gateway-fabric/blob/main/tests/reconfig/results/1.1.0/1.1.0.md#future-improvements

Reply:
Also, what's the purpose for trying the larger sleep/grace termination values?

Reply:
this is specifically for scaling down, to see if a bigger value of the delay can prevent traffic loss. we introduce that delay to allow a fronting load balancer (in this case, Google Cloud Network Load Balancer) to gracefully drain the node with a terminating pod. It might be the case that the chosen value in this test - 40s - is not enough